### DISCUSSION TITLE
Compile ANE for iOS with recommended settings does not work

### DISCUSSION THREAD

**Discussion by MobiliteDev** [airsdk/Adobe-Runtime-Support#863] [Q&A]
* Which AIR SDK did you use? harman 33.1.1.476
* Describe the affected environment: iOS ANE
* Can you reproduce it yourself ?   yes

I have to migrate my ANE from AIR 24 to AIR 33.1.
When I open the project on XCode, I have several recommended updated settings:
- min SDK to IOS 12
- compilation 

1) If I set the minimum iOS SDK to 12, the target changes from "Any iOS Device (arm64,armv4)" to "Any iOS Device (arm64)", the ane.a compiles well, but I can't generate the ANE, I have this error:
_Error: Apple App Store allows only universal applications. "myANE.a" is not a universal binary. Please change build settings in Xcode project to "Standard Architecture" to create universal library/framework._

My settings are good, it is because library is generated for arm64 only I think, is there a way to compile with min SDK iOS 12 ?

2) In the recommended settings, there is one option:
![image](https://user-images.githubusercontent.com/54275508/117774322-a53c3b00-b239-11eb-810f-e0f7fe6e4673.png)

I 'No common block' is activated, the library compiles,ths ANE too, but it is impossible to launch the app in debug mode, I have an error "Compilation failed while executing : ld64"

**Comment by dark0084** [comment]
hello.

I also have the same problem.
It happens when I don't include armv7 in the library.

https://github.com/airsdk/Adobe-Runtime-Support/issues/190

Same if I add `<option>-ios_version_min 11.0</option>` to platform.xml.
I can't request armv7 support from the third-party library I'm using now.
Anyone know how to solve it?

> Environment:
> AIR SDK 33.1.1.889
> macOS 11.16.2
> xcode 13.2.1

**Reply by ajwfrost** [reply]
@dark0084 just to check, are you creating an ANE? and by "the same problem" do you mean the message Apple App Store allows only universal applications" which stops you from creating the ANE?

@MobiliteDev are you able to drop the min SDK back to 9 so that it matches what we use for AIR?

thanks

**Reply by ajwfrost** [reply]
@dark0084 actually just looking at how this works, you've tried adding `<option>-ios_version_min 11.0</option>` to your platform XML file (within a `linkerOptions` section I assume), can you also please add `<sdkVersion>11.0</sdkVersion>` at the top level (so alongside `<linkerOptions>`)?

thanks

**Reply by dark0084** [reply]
@ajwfrost 

Hello.
Thank you for your answer.
Yes, I encountered the error message when building ane.

```
<platform xmlns="http://ns.adobe.com/air/extension/33.1">
  <sdkVersion>11.0</sdkVersion>
    <linkerOptions>
      <option>-ios_version_min 11.0</option>
    </linkerOptions>
</platform>
```

As you told me, I wrote platform.xml and built ane, and it was successful!
Thank you very much!
